operations/mimir-mixin/alerts/ingest-storage.libsonnet

```
@@ -123,12 +123,18 @@
           alert: $.alertName('IngesterFailsToProcessRecordsFromKafka'),
           'for': '5m',
           expr: |||
-            sum by (%(alert_aggregation_labels)s, %(per_instance_label)s) (
-                # This is the old metric name. We're keeping support for backward compatibility.
-              rate(cortex_ingest_storage_reader_records_failed_total{cause="server"}[1m])
-              or
-              rate(cortex_ingest_storage_reader_requests_failed_total{cause="server"}[1m])
-            ) > 0
+            (
+              sum by (%(alert_aggregation_labels)s, %(per_instance_label)s) (
+                  # This is the old metric name. We're keeping support for backward compatibility.
+                rate(cortex_ingest_storage_reader_records_failed_total{cause="server"}[1m])
+                or
+                rate(cortex_ingest_storage_reader_requests_failed_total{cause="server"}[1m])
+              ) > 0
+            )
+
+            # Tolerate failures during the forced TSDB head compaction, because samples older than the
+            # new "head min time" will fail to be appended while the forced compaction is running.
+            unless (max by (%(alert_aggregation_labels)s, %(per_instance_label)s) (cortex_ingester_tsdb_forced_compactions_in_progress) > 0)
```

```suggestion
            unless (max by (%(alert_aggregation_labels)s, %(per_instance_label)s) (max_over_time(cortex_ingester_tsdb_forced_compactions_in_progress[1m]) > 0)
```

Let's say the scrape interval is 15s, then:
- 1st scrape at 00s: `cortex_ingester_tsdb_forced_compactions_in_progress=1`
- 2nd scrape at 15s: `cortex_ingest_storage_reader_records_failed_total++`
- 3rd scrape at 45s: `cortex_ingester_tsdb_forced_compactions_in_progress=0`
- Somewhere between 45s and 1m14s999ms we evaluate the rule, and lookback says that `cortex_ingester_tsdb_forced_compactions_in_progress=0` so it _fires_ 

(Okay, it shouldn't because it should be `for:5m`, but it fires today, I guess it's because it force-compacts different tenants, anyway, this would be more correct IMO).

---

The `for: 5m` should just avoid the scenario you're describing 🤔

---

But also doing as you suggest doesn't hurt

---

Addressed in https://github.com/grafana/mimir/pull/11006/commits/42b7430b0494b6eabf6a37d421a14c8b599dcf4f